metricflow/model/transformations/dbt_to_metricflow.py

```diff
@@ -36,254 +36,266 @@ class TransformedDbtMetric:  # noqa: D
 }
 
 
-def _resolve_metric_model_ref(manifest: DbtManifest, dbt_metric: DbtMetric) -> DbtModelNode:  # noqa: D
-    if dbt_metric.model[:4] != "ref(":
-        raise RuntimeError("Can only resolve refs for ref strings that begin with `ref(`")
-    ref_parts = dbt_metric.model[4:-1].split(",")
-    target_model = None
-    target_package = None
-
-    if len(ref_parts) == 1:
-        target_model = ref_parts[0].strip(" \"'\t\r\n")
-    elif len(ref_parts) == 2:
-        target_package = ref_parts[0].strip(" \"'\t\r\n")
-        target_model = ref_parts[1].strip(" \"'\t\r\n")
-    else:
-        ref_invalid_args(dbt_metric.name, ref_parts)
-
-    node = manifest.resolve_ref(
-        target_model_name=target_model,
-        target_model_package=target_package,
-        current_project=manifest.metadata.project_id,
-        node_package=dbt_metric.package_name,
-    )
-    assert isinstance(
-        node, DbtModelNode
-    ), f"Ref `{dbt_metric.model}` resolved to {node}, which is not of type `{DbtModelNode.__name__}`"
-    return node
-
-
-def _db_table_from_model_node(node: DbtModelNode) -> str:  # noqa: D
-    return f"{node.database}.{node.schema}.{node.name}"
-
-
-def _build_dimension(name: str, dbt_metric: DbtMetric, time_dimension_stats: Dict[str, List[str]]) -> Dimension:
-    if name in time_dimension_stats.keys():
-        return Dimension(
-            name=name,
-            type=DimensionType.TIME,
-            type_params=DimensionTypeParams(
-                is_primary=dbt_metric.model in time_dimension_stats[name], time_granularity=TimeGranularity.DAY
-            ),
-        )
-    else:
-        return Dimension(
-            name=name,
-            type=DimensionType.CATEGORICAL,
+class DbtManifestTransformer:
```

Comment:
Just when I thought this module should really be a class, you made it into a class.